tiatoolbox/models/architecture/vanilla.py

```diff
@@ -268,3 +321,111 @@ def infer_batch(
             output = model(img_patches_device)
         # Output should be a single tensor or scalar
         return [output.cpu().numpy()]
+
+
+class TimmModel(CNNModel):
+    """Retrieve the tile encoder from timm.
+
+    This is a wrapper for pretrained models within timm.
+
+    Args:
+        backbone (str):
+            Model name. Currently, the tool supports following
+             model names and their default associated weights from timm.
+             - "efficientnet_b{i}" for i in [0, 1, ..., 7]
+             - "UNI"
+             - "prov-gigapath"
+        num_classes (int):
+            Number of classes output by model.
+        pretrained (bool, keyword-only):
+            Whether to load pretrained weights.
+
+    Attributes:
+        num_classes (int):
+            Number of classes output by the model.
+        pretrained (bool):
+            Whether to load pretrained weights.
+        feat_extract (nn.Module):
+            Backbone Timm model.
+        classifier (nn.Module):
+            Linear classifier module used to map the features to the
+            output.
+    """
+
+    def __init__(
+        self: TimmModel,
+        backbone: str,
+        num_classes: int = 1,
+        *,
+        pretrained: bool,
+    ) -> None:
+        """Initialize :class:`TimmModel`."""
+        ModelABC.__init__(self)
+        self.pretrained = pretrained
+        self.num_classes = num_classes
+        self.feat_extract = _get_timm_architecture(
+            arch_name=backbone, pretrained=pretrained
+        )
+
+        # Best way to retrieve channel dynamically is passing a small forward pass
+        prev_num_ch = self.feat_extract(torch.rand([2, 3, 224, 224])).shape[1]
+        self.classifier = nn.Linear(prev_num_ch, num_classes)
+
+    def forward(self: TimmModel, imgs: torch.Tensor) -> torch.Tensor:
+        """Pass input data through the model.
+
+        Args:
+            imgs (torch.Tensor):
+                Model input.
+
+        """
+        feat = self.feat_extract(imgs)
+        feat = torch.flatten(feat, 1)
+        logit = self.classifier(feat)
+        return torch.softmax(logit, -1)
+
+
+class TimmBackbone(CNNBackbone):
```

Comment:
```suggestion
class TimmBackbone(ModelABC):
```

As you are using ModelABC.__init__, you do not need CNNBackbone unless there is any other functionality you need.

Reply:
There is the `infer_batch()` method `TimmBackbone` inherits from `CNNBackbone`.